[♻️Refactor] 폴더 구조, 기능 수정

dkslel1225 opened the pull request:
<!-- [깃모지 타입] -->
<!--  ✨Feat  🐛Fix  🎨Style  🗑️Remove  ♻️Refactor  📝Docs  🚀Deploy  -->
<!--  📁Chore : 폴더 구조 변경 또는 디렉토리 작업  🔧Chore : 설정, 빌드 변경  -->

## ✨ 요약

<!-- 구현 내용or변경 사항에 대한 간단한 설명 -->
<!-- 관련 이슈는 development로 연결 또는 직접 작성( #issue, closes #issue ) -->
- ♻️
  - Calendar: fragments 폴더 내부로 구성요소 파일 분리(arrow-button, 요일, 이번달 아닌 날짜, days-of-month)
  - Calendar: useCalendarStore: 캘린더 내부 libs로 위치 변경
  - reservation form: 모달 파일 분리(success,need login)
- ✨
  - 비로그인 상태에서 ‘예약하기’클릭 → 모달에서 ‘로그인하기’클릭 → 로그인 성공 → 상세페이지로 다시 리다이렉트 (세션스토리지 사용하여 구현)
- 🎨
  - 모달 오픈 애니메이션 추가
- 🐛
  - 모바일 - 시간대만 선택해도 예약신청으로 넘어가는 에러 해결
 
### 추가 작업 내용
- 체험상세페이지에서 로그인,로그아웃,회원가입 시 체험상세페이지로 다시 리다이렉트
    - 세션스토리지 활용
    - useSavePathActivityId에서 경로 저장
    - useRedirectAfterSuccess에서 리다이렉트 경로 결정
- 모달 - warning 이미지 미리 로드
    - 모달 첫 렌더링 시 약간 버벅이는 느낌이 있어서 적용함
    - basic-modal이고 모달타입이 warning이면, 이미지 로드 완료까지 closeModal(esc or click..) 실행 막음
    
- 오픈 리다이렉트(Open Redirect) 취약점 문제 방지(코드래빗 리뷰) 
- 21일 회의 피드백 반영
    - star이미지 재다운로드 및 폴더 내부로 파일 이동
    - 상세페이지-리뷰 점수-소수점 1자리 추가: toFixed(1)
## 📝 상세 내용

<!-- 구현 내용에 대한 자세한 설명 -->

## 🖼️ 스크린샷

<!-- UI 변경이 있는 경우 변경 전/후 스크린샷 -->

## ✅ 체크리스트

- [ ] 브랜치 네이밍 컨벤션을 준수했습니다
- [ ] 커밋 컨벤션을 준수했습니다
- [ ] 코드가 프로젝트의 스타일 가이드라인을 준수합니다

## 💡 참고 사항


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - 로그인 필요 경고 모달 및 예약 완료 확인 모달 추가
  - 예약용 SubmitButton 컴포넌트 도입
  - 로그인 후 복귀 처리를 위한 세션 기반 리다이렉트 지원 추가

- Improvements
  - 모바일(isMobile)/데스크탑에 따른 예약·모달 흐름 및 단계 전환 개선
  - 캘린더 요일·월명 상수화(MONTH_NAME, DAYS) 및 너비 적용 일관화
  - 로그인/회원가입/로그아웃 시 경로 저장 후 안전한 리다이렉트 처리

- UI
  - 모달에 애니메이션(페이드·스케일) 적용 및 ESC 동작 안정화(아이콘 로드 대기)
  - 별 아이콘 자원 교체(활성화된 스타)
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
🎨 style: globals.css - 브랜드 컬러 수정

Commit:
pull origin dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev'

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
♻️ refactor: 캘린더 폴더구조 리펙토링
- fragments 폴더 내부로 구성요소 파일 분리
- useCalendarStore: 캘린더 내부 libs로 위치 변경

Commit:
✨ feat: 로그인 성공 후 상세페이지로 다시 리다이렉트
-상세페이지에서 로그인 페이지로 넘어간 경우에 해당
-pathname, 세션스토리지 활용

Commit:
🎨 style: basic-modal 오픈 애니메이션 추가

Commit:
🐛 ♻️ fix: 모바일 - 시간대만 선택해도 예약신청으로 넘어가는 에러 해결

Commit:
♻️ refactor: 체험예약폼 관련 - 모달 파일 분리(success,need login)

Commit:
🐛 fix: submit-button 로직 수정

Commit:
🎨 style: star 이미지 다시 다운로드

Commit:
♻️ refactor: StarImage 폴더 내부로 이동

Commit:
✨ feat: 리뷰 점수 - 소수점 추가 (toFixed)

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' into 174-refactor/refactoring

Commit:
Merge branch 'dev' of https://github.com/codeit-team6/nomadia into dev

Commit:
Merge branch 'dev' into 174-refactor/refactoring

0Hyunn reviewed (approved):
작업 내역이 많네요! 고생하셨습니다~~ 
내일 qa때 마저 수정하신 거 공유 해주시면 감사하겠습니다!! 🐰🐰

Commit:
✨ feat: 체험상세페이지에서 로그인,로그아웃 시 다시 리다이렉트
- useSavePathActivityId: 로그인,로그아웃,회원가입으로 넘어갈때 체험상세페이지일 경우 경로 미리 저장
- useRedirectAfterSuccess: 요청 성공 후(로그인,로그아웃), 리다이렉트 관련 로직

Commit:
✨ feat: 모달 - warning 이미지 미리 로드
- basic-modal의 경우 이미지 로드 완료까지 closeModal(esc or click..) 막음

Commit:
🐰refactor: 코드래빗 리뷰 반영
